phoenicis-tools/src/main/java/org/phoenicis/tools/lnk/LnkParser.java

```diff
@@ -0,0 +1,156 @@
+     * @return true or false
+     */
+    private boolean isLnkLocal(byte[] rawLnkContent) {
+        final int fileLocationInfoFlag = rawLnkContent[FILE_LOCATION_INFO_FLAG_OFFSET_OFFSET];
+        return (fileLocationInfoFlag & 2) == 0;
+    }
+
+    /**
+     * Parse the acutal content of the shortcut
+     *
+     * @param rawLnkShortcutByteArrayWithoutHeader Raw lnk file content
+     * @param isLocal Search for local or network filename
+     */
+    private String parseLnkContent(byte[] rawLnkShortcutByteArrayWithoutHeader, boolean isLocal) {
+        final int finalNameOffset = rawLnkShortcutByteArrayWithoutHeader[FINALNAME_OFFSET_OFFSET];
+
+        final String finalName = BytesUtilities.getNullDelimitedString(rawLnkShortcutByteArrayWithoutHeader,
+                finalNameOffset);
+
+        if (isLocal) {
+            final int basenameOffset = rawLnkShortcutByteArrayWithoutHeader[BASENAME_OFFSET_OFFSET];
+            final String basename = BytesUtilities.getNullDelimitedString(rawLnkShortcutByteArrayWithoutHeader,
+                    basenameOffset);
+            return basename + finalName;
+        } else {
+            final int networkVolumeTableOffset = rawLnkShortcutByteArrayWithoutHeader[NETWORK_VOLUME_TABLE_OFFSET_OFFSET];
+            final int shareNameOffset = rawLnkShortcutByteArrayWithoutHeader[networkVolumeTableOffset
+                    + SHARE_NAME_OFFSET_OFFSET] + networkVolumeTableOffset;
+            final String shareName = BytesUtilities.getNullDelimitedString(rawLnkShortcutByteArrayWithoutHeader,
+                    shareNameOffset);
+            return shareName + "\\" + finalName;
+        }
+    }
+
+    /**
+     * Fetches the start of the .lnk file once the header is removed
+     *
+     * @param rawLnkContent The content of the .lnk
+     * @param dataFlags The dataflags that will be used to determine if there is a shell section
+     * @return The offset where the sortcuts file starts
+     */
+    private int fetchFileStart(byte[] rawLnkContent, LnkData dataFlags) {
+        return LNK_HEADER_SIZE + fetchShellLength(rawLnkContent, dataFlags);
+    }
+
+    /**
+     * Fetches shell section length
+     *
+     * @param rawLnkContent raw .lnk content
+     * @param dataFlags {@link LnkData to fetch shell}
+     * @return the size of the section
+     */
+    private int fetchShellLength(byte[] rawLnkContent, LnkData dataFlags) {
+        int shellLength = 0;
+        if (dataFlags.hasShell()) {
+            final int lengthMarkerSize = 2;
+            shellLength = BytesUtilities.bytes2short(rawLnkContent, LNK_HEADER_SIZE) + lengthMarkerSize;
+        }
+
+        return shellLength;
+    }
+
+    /**
+     * Fetches the file attributes
+     *
+     * @param rawLnkContent raw .lnk content
+     * @return a {@link LnkFileAttribute} property
+     */
+    private LnkFileAttribute fetchFilesAttributes(byte[] rawLnkContent) {
+        return new LnkFileAttribute(rawLnkContent);
+    }
+}
```

Comment:
newline at EOF (there's a setting in IntelliJ IDEA)

Reply:
Done